//查询参数
message db_query_param
{
	//等同于MYSQL_TYPE_XXX
	required int32 param_type = 1;
	//根据TYPE_XXX判断类型
	required bytes values = 2;
};

//一个构造查询信息的结构
message db_query
{
	//数据库连接标志
	//使用哪个数据库连接
	required string conn = 1;

	//查询的sql 
	//select * from tbl where a = ? and b = ?
	//使用参数绑定
	required string query = 2;

	//参数绑定的参数 按照sql的顺序传入
	repeated db_query_param params = 3;
};


message db_field
{
	//field的名称
	required string name = 1;

	//等同于MYSQL_TYPE_XXXX
	required int32 type = 2;
};

message db_fields
{
	repeated db_field fields = 1;
};

message db_row_value
{
	required bytes value = 1;
	required bool is_null = 2 [default = false];
	required bool is_unsigned = 3 [default = false];
};

message db_row
{
	repeated db_row_value column = 1;
};

message db_rows
{
	repeated db_row rows = 1;
};
message db_error
{
	required bool is_mysql_error = 1;
	required int32 error_code = 2;
	required string reason = 3;
};
message db_query_result
{
	//数据库字段信息
	optional db_fields fields = 1;
	
	//update delete insert影响的行数
	optional uint64 affected_rows = 2;

	//表示查询的结果row
	optional db_rows rows = 3;

	//是否操作成功
	required bool success = 4;
	
	//数据库记录错误信息
	optional db_error error = 5;
};
